macos/ds_store.ksy

```diff
@@ -0,0 +1,202 @@
+meta:
+  id: ds_store
+  title: macOS '.DS_Store' format
+  license: MIT
+  ks-version: 0.8
+  encoding: UTF-8
+  endian: be
+doc: |
+  Apple macOS '.DS_Store' file format.
+doc-ref: |
+  https://en.wikipedia.org/wiki/.DS_Store
+  https://metacpan.org/pod/distribution/Mac-Finder-DSStore/DSStoreFormat.pod
+  https://0day.work/parsing-the-ds_store-file-format
+seq:
+  - id: alignment_header
+    contents: [0x00, 0x00, 0x00, 0x01]
+  - id: buddy_allocator_header
+    type: buddy_allocator_header
+instances:
+  buddy_allocator_body:
+    pos: buddy_allocator_header.ofs_bookkeeping_info_block + 4
+    size: buddy_allocator_header.len_bookkeeping_info_block
+    type: buddy_allocator_body
+types:
+  buddy_allocator_header:
+    seq:
+      - id: magic
+        contents: ["Bud1"]
+        doc: Magic number 'Bud1'
+      - id: ofs_bookkeeping_info_block
+        type: u4
+      - id: len_bookkeeping_info_block
+        type: u4
+      - id: copy_ofs_bookkeeping_info_block
+        type: u4
+        doc: Needs to match 'offset_bookkeeping_info_block'
+      - size: 16
+        doc: Unused block
+  buddy_allocator_body:
+    seq:
+      - id: num_blocks
+        type: u4
+        doc: Number of blocks in the allocated-blocks list
+      - size: 4
+        doc: Unknown field
+      - id: block_addresses
+        type: block_descriptor
+        repeat: expr
+        repeat-expr: 256
+        doc: Addresses of the different blocks
+      - id: num_directories
+        type: u4
+        doc: Indicates the number of directory entries
+      - id: directory_entries
+        type: directory_entry
+        repeat: expr
+        repeat-expr: num_directories
+        doc: Each directory is an independent B-tree
+      - id: free_lists
+        type: free_list
+        repeat: expr
+        repeat-expr: 32
```

Comment:
magic constant can be moved into an own instance

Reply:
Done.